### PR TITLE
MEN-4332: Remove sensitive information from logs

### PR DESCRIPTION
**Pull request by alfrunes**
Removes device type, artifact- and release name from log entries. I also did some minor lint-check fixes.

- [x] Blocked by mendersoftware/deployments-enterprise#291

### CODE REVIEW

File: api/http/api_deployments.go [hunk 1]
```diff
@@ -119,9 +119,20 @@ func (d *DeploymentsApiHandlers) GetReleases(w rest.ResponseWriter, r *rest.Requ
...
 
-	name := r.URL.Query().Get("name")
+	q := r.URL.Query()
+	name := q.Get("name")
 
 	if name != "" {
+		defer func() {
+			var redacted bool
+			if q.Get("name") != "" {
+				q.Set("name", "REDACTED")
```
Comment (merlin-northern) [inline api/http/api_deployments.go:129]: I will keep repeating that, sorry, could we please have a const "REDACTED"?

File: api/http/api_deployments.go [hunk 2]
```diff
@@ -965,8 +990,19 @@ func parseEpochToTimestamp(epoch string) (time.Time, error) {
 func (d *DeploymentsApiHandlers) LookupDeployment(w rest.ResponseWriter, r *rest.Request) {
 	ctx := r.Context()
 	l := requestlog.GetRequestLogger(r)
+	q := r.URL.Query()
+	defer func() {
+		var redacted bool
+		if search := q.Get("search"); search != "" {
+			q.Set("search", "REDACTED")
+			redacted = true
```
Comment (merlin-northern) [inline api/http/api_deployments.go:997]: why do we need the flag `redacted` here?

File: api/http/api_deployments.go [hunk 3]
```diff
@@ -352,7 +352,6 @@ func (d *DeploymentsApiHandlers) newImageWithContext(ctx context.Context, w rest
 
 	// parse multipart message
 	multipartUploadMsg, err := d.ParseMultipart(formReader)
-	defer r.MultipartForm.RemoveAll()
```
Comment (merlin-northern) [inline api/http/api_deployments.go:344]: I am not sure about that, why are we dropping that?

**Review by merlin-northern — COMMENTED** [review]
left some comments, not critical, I am not sure about one thing, otherwise looks good.

File: api/http/api_deployments.go (hunk 3 above)
Comment (alfrunes) [reply api/http/api_deployments.go:344]: This statement is a NOOP. We're not calling `r.ParseMultipartForm` anymore.

File: api/http/api_deployments.go (hunk 2 above)
Comment (alfrunes) [reply api/http/api_deployments.go:997]: We don't :)

**Review by merlin-northern — APPROVED** [review]
lgtm!

**Review by tranchitella — APPROVED** [review]

**Pull request merged**